<a href="https://colab.research.google.com/github/dacdemon/ITD/blob/main/INTEGRADOR%20CORIA%20DAMIAN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# TP Integrador 1
# Análisis Exploratorio y Predictivo (Python)
##Alumno: Damián Coria
##Comisión: 2

El Ministerio de Energía busca reducir progresivamente la dependencia de los combustibles fósiles y aumentar la participación de fuentes renovables en la matriz energética.

Ante esta situación, se solicita realizar un análisis de datos históricos de distintos países con el objetivo de estudiar la evolución de las principales fuentes de energía renovable, como la energía hidroeléctrica, solar y eólica.

El propósito del análisis será identificar tendencias, comparar el crecimiento de estas tecnologías y evaluar cuál de ellas presenta mejores condiciones para acompañar una transición hacia una matriz energética con menor dependencia de combustibles fósiles.

A partir de los resultados obtenidos, se elaborará una conclusión basada en los datos analizados.

In [ ]:
# Importamos las librerias necesarias para esta praactica (pandas, matplot y seaborn).

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Cargamos los archivos csv desde un repositorio web.

url = "https://raw.githubusercontent.com/dacdemon/ITD/refs/heads/main/renewable_energy_by_country.csv"
df = pd.read_csv(url)


# El DataFrame original es muy grande, por lo que se realiza una exploración inicial para conocer sus dimensiones, las columnas disponibles y el período de los datos. A partir de este análisis se seleccionarán las variables y registros más relevantes, respetando los requisitos mínimos de la consigna: 3000 filas y 5 columnas.

In [ ]:
# Se crea una función de análisis para reutilizarla y verificar los cambios realizados en el DataFrame.

def analizar_dataset(df):

    # Dimensiones:

    print("\nDimensiones:")
    dimensiones = df.shape
    print(dimensiones[0], "Filas")
    print(dimensiones[1], "Columnas")

    # columnas disponibles:

    print("\nColumnas:")
    print(df.columns)

    # se analiza dos columnas claves: paises y año

    print("\nCantidad de países:", df["country"].nunique())
    print("Año inicial:", df["year"].min())
    print("Año final:", df["year"].max())

analizar_dataset(df)


Dimensiones:
17265 Filas
29 Columnas

Columnas:
Index(['country', 'year', 'iso_code', 'population', 'gdp',
       'primary_energy_consumption', 'energy_per_capita',
       'fossil_share_energy', 'fossil_fuel_consumption',
       'electricity_generation', 'renewables_share_energy',
       'renewables_share_elec', 'renewables_consumption',
       'renewables_electricity', 'hydro_electricity', 'hydro_share_elec',
       'hydro_share_energy', 'solar_electricity', 'solar_share_elec',
       'solar_share_energy', 'wind_electricity', 'wind_share_elec',
       'wind_share_energy', 'other_renewable_electricity',
       'other_renewables_share_elec', 'other_renewables_share_energy',
       'renewables_electricity.1', 'renewables_share_elec.1',
       'renewables_share_energy.1'],
      dtype='object')

Cantidad de países: 220
Año inicial: 1900
Año final: 2025


# Se seleccionaron estas columnas porque están directamente relacionadas con el objetivo del análisis. Permiten comparar la participación de los combustibles fósiles con distintas fuentes de energía renovable, analizar su evolución a lo largo del tiempo y estudiar las relaciones entre las variables.

In [ ]:
# columnas elegidas.

variables = [
    "country",
    "year",
    "fossil_share_energy",
    "hydro_share_energy",
    "solar_share_energy",
    "wind_share_energy",
    "other_renewables_share_energy",
    "primary_energy_consumption"]

df_analisis = df[variables]

print(df_analisis.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17265 entries, 0 to 17264
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   country                        17265 non-null  object 
 1   year                           17265 non-null  int64  
 2   fossil_share_energy            4459 non-null   float64
 3   hydro_share_energy             4459 non-null   float64
 4   solar_share_energy             4459 non-null   float64
 5   wind_share_energy              4459 non-null   float64
 6   other_renewables_share_energy  4459 non-null   float64
 7   primary_energy_consumption     10355 non-null  float64
dtypes: float64(6), int64(1), object(1)
memory usage: 1.1+ MB
None


# En el análisis anterior se observó una gran cantidad de valores nulos.
#Como el dataset comienza en 1900, se plantea la hipótesis de que estos valores pueden deberse a la falta de registros de algunas tecnologías energéticas en los años más antiguos.
# Por este motivo, se analizará desde qué año existe información completa para las variables seleccionadas.

In [ ]:
df_analisis = df_analisis.dropna()

print("Dimensiones:", df_analisis.shape)
print("Cantidad de países:", df_analisis["country"].nunique())
print("Período:", df_analisis["year"].min(), "-", df_analisis["year"].max())

Dimensiones: (4459, 8)
Cantidad de países: 79
Período: 1965 - 2024


# La hipótesis planteada anteriormente era correcta: los datos completos para las variables seleccionadas se concentran en años más recientes, desde 1965 hasta 2024.
# De esta manera, se obtiene un DataFrame de 4459 filas y 8 columnas, que cumple con los requisitos mínimos de la consigna y mantiene datos relevantes para el análisis.

In [ ]:
# Procedemos a reducir el DataFrame original y lo reemplazamos por el conjunto de datos elejido

columnas_seleccionadas = [
    "country",
    "year",
    "fossil_share_energy",
    "hydro_share_energy",
    "solar_share_energy",
    "wind_share_energy",
    "other_renewables_share_energy",
    "primary_energy_consumption"]

# Se conservan las columnas seleccionadas y se eliminan las filas con valores nulos

df = df[columnas_seleccionadas].dropna()

# Utilizamos la funcion previamente creada para analizar como quedo nuestro dataframe

analizar_dataset(df)


Dimensiones:
4459 Filas
8 Columnas

Columnas:
Index(['country', 'year', 'fossil_share_energy', 'hydro_share_energy',
       'solar_share_energy', 'wind_share_energy',
       'other_renewables_share_energy', 'primary_energy_consumption'],
      dtype='object')

Cantidad de países: 79
Año inicial: 1965
Año final: 2024


# Una vez preparado el dataset y seleccionados los datos relevantes para nuestro análisis, se inicia el desarrollo del Trabajo Práctico Integrador.

#<u><b> PARTE 1: INGESTA Y ESTRUCTURA </b></u>
---
---
##Identificar el tipo de variables (numéricas, categóricas, fechas).

In [ ]:
# Analizamos los tipos de datos de las variables seleccionadas

display(df.dtypes)

,0
country,object
year,int64
fossil_share_energy,float64
hydro_share_energy,float64
solar_share_energy,float64
wind_share_energy,float64
other_renewables_share_energy,float64
primary_energy_consumption,float64


# Tipos de variables

##Country: Variable categórica. Identifica el país al que corresponde cada registro.

##Year: Variable temporal representada mediante números enteros. Indica el año de cada registro.

##Fossil_share_energy: Variable numérica continua. Representa la participación porcentual de los combustibles fósiles en el consumo de energía primaria.

##Hydro_share_energy: Variable numérica continua. Representa la participación porcentual de la energía hidroeléctrica.

##Solar_share_energy: Variable numérica continua. Representa la participación porcentual de la energía solar.

##Wind_share_energy: Variable numérica continua. Representa la participación porcentual de la energía eólica.

##Other_renewables_share_energy: Variable numérica continua. Representa la participación porcentual de otras fuentes renovables.

##Primary_energy_consumption: Variable numérica continua. Representa el consumo de energía primaria.

In [ ]:
# Renombramos las columnas para facilitar su lectura durante el análisis

df = df.rename(columns={
    "country": "Pais",
    "year": "Año",
    "fossil_share_energy": "Porcentaje_Fosil",
    "hydro_share_energy": "Porcentaje_Hidroelectrica",
    "solar_share_energy": "Porcentaje_Solar",
    "wind_share_energy": "Porcentaje_Eolica",
    "other_renewables_share_energy": "Porcentaje_Otras_Renovables",
    "primary_energy_consumption": "Consumo_Energia_Primaria"})

# Verificamos los nuevos nombres

print(df.columns)

# volvemos a Verificar los tipos de datos

display(df.dtypes)

Index(['Pais', 'Año', 'Porcentaje_Fosil', 'Porcentaje_Hidroelectrica',
       'Porcentaje_Solar', 'Porcentaje_Eolica', 'Porcentaje_Otras_Renovables',
       'Consumo_Energia_Primaria'],
      dtype='object')


,0
Pais,object
Año,int64
Porcentaje_Fosil,float64
Porcentaje_Hidroelectrica,float64
Porcentaje_Solar,float64
Porcentaje_Eolica,float64
Porcentaje_Otras_Renovables,float64
Consumo_Energia_Primaria,float64


# Justificación de la variable Año

## La variable `Año` se mantiene como int64 porque el dataset registra únicamente años completos y no fechas específicas. Este formato permite ordenar, filtrar y analizar la evolución de los datos a lo largo del tiempo sin necesidad de convertirla a datetime.

#<u><b> PARTE 2: LIMPIEZA Y ESTADÍSTICA </b></u>
---
---
## Tratar valores nulos y duplicados.
## Descripción: Generar un resumen estadístico con  `describe()` e interpretar la media y la dispersión de los datos.

## Antes de comenzar con el análisis estadístico, se verifica la calidad del DataFrame buscando valores faltantes y registros duplicados que puedan afectar los resultados. Aunque durante la preparación del dataset ya utilizamos `dropna()` para eliminar los valores faltantes, realizaremos una nueva verificación para confirmar que no queden datos faltantes ni filas duplicadas.

In [ ]:
# Verificamos valores faltantes

print("Valores faltantes:")
display(df.isnull().sum())

# Verificamos registros duplicados

print("\nRegistros duplicados:")
print(df.duplicated().sum())

Valores faltantes:


,0
Pais,0
Año,0
Porcentaje_Fosil,0
Porcentaje_Hidroelectrica,0
Porcentaje_Solar,0
Porcentaje_Eolica,0
Porcentaje_Otras_Renovables,0
Consumo_Energia_Primaria,0



Registros duplicados:
0


# Análisis de valores faltantes y duplicados

## Luego de realizar la verificación, se observa que el DataFrame **no presenta valores faltantes ni registros duplicados**.

## Esto confirma que el proceso de limpieza realizado anteriormente fue correcto, por lo que **no es necesario realizar nuevas modificaciones sobre los datos**.

# Análisis estadístico

## Una vez comprobado que el DataFrame no contiene valores faltantes ni registros duplicados, se realiza un **análisis estadístico de las variables numéricas** utilizando `describe()`.

## Este análisis permite observar medidas como la **media, la desviación estándar, los valores mínimos y máximos, y los cuartiles** de cada variable.


In [ ]:
df.describe()

,Año,Porcentaje_Fosil,Porcentaje_Hidroelectrica,Porcentaje_Solar,Porcentaje_Eolica,Porcentaje_Otras_Renovables,Consumo_Energia_Primaria
count,4459.000000,4459.000000,4459.000000,4459.000000,4459.000000,4459.000000,4459.000000
mean,1995.727742,86.371914,8.591933,0.231444,0.576266,0.896672,1336.736913
std,17.056720,15.616671,12.420324,0.830977,1.955603,2.506629,3762.640049
min,1965.000000,13.874000,0.000000,0.000000,0.000000,0.000000,0.097000
25%,1981.000000,80.544000,0.426500,0.000000,0.000000,0.000000,136.192000
50%,1996.000000,91.794000,3.602000,0.000000,0.000000,0.053000,349.032000
75%,2010.000000,97.993500,11.363000,0.004000,0.059000,0.628000,1004.991500
max,2024.000000,100.000000,71.417000,9.881000,26.085000,31.244000,48987.102000


# INTERPRETACIÓN DEL ANÁLISIS ESTADÍSTICO

## El análisis muestra que los combustibles fósiles tienen una participación media del **86,37 %**, con una desviación estándar de **15,62 %**. Esto indica que, aunque existen diferencias entre países y años, los combustibles fósiles presentan una participación elevada en gran parte de los registros analizados.

## Entre las energías renovables, la **hidroeléctrica** presenta la mayor participación media, con **8,59 %**, y también la mayor dispersión, con una desviación estándar de **12,42 %**. Esto indica que su participación varía considerablemente entre los distintos países y años.

## La energía **eólica** presenta una media de **0,58 %** y una desviación estándar de **1,96 %**, mientras que la energía **solar** tiene una media de **0,23 %** y una desviación estándar de **0,83 %**. Además, ambas presentan una mediana de **0 %**, lo que indica que en al menos la mitad de los registros históricos su participación fue nula.

## Las **otras energías renovables** presentan una media de **0,90 %** y una desviación estándar de **2,51 %**, mostrando también una participación relativamente baja y variable.

## Por último, el **Consumo_Energia_Primaria** presenta una gran dispersión, con una media de **1336,74** y una desviación estándar de **3762,64**, lo que evidencia importantes diferencias en el nivel de consumo energético entre los países y años analizados.

#<u><b> PARTE 3: ANÁLISIS VISUAL </b></u>
---
---
## Una vez analizadas las características estadísticas del dataset, se comienza con el análisis visual de los datos.
## El objetivo será identificar patrones y tendencias que permitan comparar la participación de las distintas fuentes de energía renovable y su relación con el uso de combustibles fósiles.